### PR TITLE
[8.0] port of module logistic_requisition

### PR DESCRIPTION
**Pull request by yvaucher**
Here is a first PR porting the code of logistic_requisition to 8.0 api and to new 8.0 workflow

Some tests are commented as they need to be adapted to new 8.0 workflow.

It also:
- breaks link between logistic requisition source and sale order line to simplify the code and to add flexibility
- ports webkit report to Qweb
- removes dependancy on transport_plan

depends on
- [x] https://github.com/OCA/sale-workflow/pull/49
- [x] https://github.com/OCA/purchase-workflow/pull/26
- [x] https://github.com/OCA/purchase-workflow/pull/36

### CODE REVIEW

**Comment by gurneyalex** [comment]
could you rename ./logistic_requisition/tests/test_mutlicurrency_update_po_line.py to ./logistic_requisition/tests/test_multicurrency_update_po_line.py?

**Comment by yvaucher** [comment]
ok mu **lt** i

**Comment by jgrandguillaume** [comment]
Hi,

The button "button_create_cost_estimate" from the LR form is not working properly. 

a) Create a LR with 2 lines, confirm LR,  source lines with whatever you want (dispatch is the easiest way)
b) Mark all LR line as "sourced"
c) Try to create a coste estimate using the button on LR form => not working. BUT is I open the line using "view lines" button on LR form, select them all, chose "More->create cost estimate" it's working.

Regards,

Joël

**Comment by gurneyalex** [comment]
activity_code, account_code: this is something customer specific, which should not be in this project -> to remove

File: logistic_requisition/model/logistic_requisition.py [hunk 1]
```diff
@@ -0,0 +1,1067 @@
...
+        default='draft'
+    )
+    currency_id = fields.Many2one(
+        related='requisition_id.currency_id',
+        comodel_name='res.currency',
+        string='Currency',
+        readonly=True)
+    note = fields.Text('Notes')
+    activity_code = fields.Char('Activity Code', size=32)
+    account_code = fields.Char('Account Code', size=32)
```
Comment (yvaucher) [inline logistic_requisition/model/logistic_requisition.py]: activity_code and account_code to remove as it is too specific

File: logistic_requisition/test/logistic_requisition_report_test.yml [hunk 2]
```diff
@@ -10,8 +10,10 @@
...
     pricelist_id: product.list0
     line_ids:
-      - product_id: product.product_product_7
+      - product_id: product.product_product_32
+        description: HEAD Headset standard
         requested_qty: 100
+        requested_uom_id: product.product_uom_unit
         date_delivery: !eval "time.strftime('%Y-%m-%d')"
         logistic_user_id: base.user_demo
         account_code: 'Shazam!'
```
Comment (yvaucher) [inline logistic_requisition/test/logistic_requisition_report_test.yml]: to remove from test as well

File: logistic_requisition/tests/test_mto_workflow.py [hunk 3]
```diff
@@ -65,41 +63,48 @@ class test_mto_workflow(common.TransactionCase):
...
-        self.line = {
-            'product_id': self.product_7,
+        self.line_vals = {
+            'product_id': self.product_32.id,
+            'description': "[HEAD] Headset standard",
             'requested_qty': 100,
-            'requested_uom_id': self.product_uom_pce,
+            'requested_uom_id': self.product_uom_pce.id,
             'date_delivery': time.strftime(D_FMT),
             'account_code': 'a code',
```
Comment (yvaucher) [inline logistic_requisition/tests/test_mto_workflow.py]: to remove from test as well

File: logistic_requisition/tests/test_mto_workflow.py [hunk 4]
```diff
@@ -183,73 +199,67 @@ def _check_sale_line(self, sale_line):
...
         # add a line in the logistic requisition
         line2 = {
-            'product_id': self.product_8,
+            'product_id': self.product_25.id,
+            'description': "[LAP-E5] Laptop E5023",
             'requested_qty': 200,
-            'requested_uom_id': self.product_uom_pce,
+            'requested_uom_id': self.product_uom_pce.id,
             'date_delivery': time.strftime(D_FMT),
             'account_code': 'a code',
```
Comment (yvaucher) [inline logistic_requisition/tests/test_mto_workflow.py]: to remove from test as well

**Comment by yvaucher** [comment]
@jgrandguillaume button fixed

**Comment by yvaucher** [comment]
@gurneyalex extra fields moved

**Comment by gurneyalex** [comment]
sourcing from PR is completely broken. I'm on it (and adding the missing test)

**Comment by yvaucher** [comment]
@gurneyalex warning it seams you added few temp files in last commit

**Comment by gurneyalex** [comment]
:+1: I spent lots of time reviewing this and testing, and I think most of the ground is covered. What remains can certainly be processed using bug fixing PRs. Waiting  for Travis to flag my last commit as green and I'll merge.

**Comment by yvaucher** [comment]
agreed :+1:

File: logistic_requisition/model/requisition_cancel.py~HEAD [hunk 5]
```diff
@@ -1,5 +1,5 @@
 # -*- coding: utf-8 -*-
```
Comment (yvaucher) [inline logistic_requisition/model/requisition_cancel.py~HEAD]: some cleaning to do this file shouldn't be there
..stic_requisition/model/requisition_cancel.py~HEAD

**Comment by guewen** [comment]
Applause!

**Comment by gurneyalex** [comment]
:lollipop:

**Comment by lepistone** [comment]
Thanks for taking care of that! I :+1: the merge FWIW.

**Comment by yvaucher** [comment]
:tada:

**Pull request merged**